mllib/src/main/scala/org/apache/spark/mllib/linalg/Matrices.scala

```diff
@@ -278,7 +278,8 @@ class DenseMatrix @Since("1.3.0") (
   }
 
   override def hashCode: Int = {
-    com.google.common.base.Objects.hashCode(numRows : Integer, numCols: Integer, toArray)
+    val state = Seq(numRows, numCols, Arrays.hashCode(values), isTransposed.hashCode)
```

Comment:
I've asked on breeze mailing list and waiting for their reply.

Also I checked breeze CSCMatrix '==' method implementation , it seems '==' uses  the input data we passed from SparseMatrix and just hashCode is not implemented. I think this will guarantee  that my hashCode implementation will be consistent with Breeze '=='.
All MLlib equals hashCode test cases are passing now